bot/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+use abstract_client::{AbstractClient, AccountSource, Environment};
+use cosmos_sdk_proto::{
+    cosmwasm::wasm::v1::{query_client::QueryClient, QueryContractsByCodeRequest},
+    traits::Message as _,
+};
+use cw_orch::{
+    anyhow,
+    daemon::{networks::OSMO_5, queriers::Authz, Daemon},
+    environment::TxHandler,
+    prelude::*,
+    tokio::runtime::Runtime,
+};
+use log::{log, Level};
+use osmosis_std::types::{
+    cosmos::{
+        authz::v1beta1::GenericAuthorization,
+        bank::v1beta1::{MsgSend, SendAuthorization},
+    },
+    osmosis::{
+        concentratedliquidity::v1beta1::{
+            MsgAddToPosition, MsgCollectIncentives, MsgCollectSpreadRewards, MsgCreatePosition,
+            MsgWithdrawPosition,
+        },
+        gamm::v1beta1::MsgSwapExactAmountIn,
+    },
+};
+use savings_app::{
+    msg::{AppExecuteMsg, ExecuteMsg},
+    AppInterface,
+};
+use std::collections::HashSet;
+use tonic::transport::Channel;
+
+use abstract_app::{
+    abstract_core::version_control::ModuleFilter,
+    abstract_interface::VCQueryFns,
+    objects::{
+        module::{ModuleId, ModuleStatus},
+        namespace::ABSTRACT_NAMESPACE,
+    },
+};
+use savings_app::contract::APP_ID;
+
+use std::iter::Extend;
+
+const AUTHORIZATION_URLS: &[&str] = &[
+    MsgCreatePosition::TYPE_URL,
+    MsgSwapExactAmountIn::TYPE_URL,
+    MsgAddToPosition::TYPE_URL,
+    MsgWithdrawPosition::TYPE_URL,
+    MsgCollectIncentives::TYPE_URL,
+    MsgCollectSpreadRewards::TYPE_URL,
+];
+
+/// entrypoint for the bot
+pub fn cron_main() -> anyhow::Result<()> {
+    let rt = Runtime::new()?;
+    let mut daemon = Daemon::builder()
+        .handle(rt.handle())
+        .chain(OSMO_5)
+        .build()?;
+
+    let abstr = AbstractClient::new(daemon.clone())?;
+    let _module_id = ModuleId::try_from(APP_ID)?;
+
+    let saving_modules = abstr.version_control().module_list(
+        Some(ModuleFilter {
+            namespace: Some(ABSTRACT_NAMESPACE.to_string()),
+            name: Some("savings-app".to_string()),
+            version: None,
+            status: Some(ModuleStatus::Registered),
+        }),
+        None,
+        None,
+    )?;
+
+    let mut contract_instances_to_check: HashSet<(String, Addr)> = HashSet::new();
+
+    for app_info in saving_modules.modules {
+        let code_id = app_info.module.reference.unwrap_app()?;
+        let mut contract_addrs = rt
+            .handle()
+            .block_on(fetch_instances(daemon.channel(), code_id))?;
```

Comment:
Is it desired behavior to support multi-vestion savings-app? Authorizations and API can change from version to version!

Reply:
@CyberHoward

Reply:
We can add that logic when we release the next version